### PR TITLE
Mapping azure_sentinel UDI connector

### PR DESCRIPTION
**Pull request by JingqiuDu**

### CODE REVIEW

File: stix_shifter_modules/azure_sentinel/stix_translation/json/to_stix_map.json [hunk 1]
```diff
@@ -30,19 +30,32 @@
...
+      {
+      "key": "x-oca-event.category",
+      "object": "event"
+      }
+    ],
     "closedDateTime": {
-      "key": "x-msazure-sentinel-alert.closedDateTime",
-      "object": "alert"
+      "key": "x-ibm-finding.closedDateTime",
+      "object": "finding"
```
Comment (mdazam1942) [inline stix_shifter_modules/azure_sentinel/stix_translation/json/to_stix_map.json:45]: I think `closedDateTime` should be mapped to `last_observed`
```
"closedDateTime": {
      "key": "last_observed",
      "cybox": false
    }
```

File: stix_shifter_modules/azure_sentinel/stix_translation/json/to_stix_map.json [hunk 2]
```diff
@@ -67,13 +80,17 @@
...
+      {
+        "key": "x-oca-eventg.created",
+        "object": "event"
       }
     ],
     "description": {
-      "key": "x-msazure-sentinel-alert.description",
-      "object": "alert"
+      "key": "x-ibm-finding.description",
+      "object": "finding"
```
[the remark this answers is not in the document]
Comment (delliott90) [reply stix_shifter_modules/azure_sentinel/stix_translation/json/to_stix_map.json:93]: It isn't in the definition but I think it should be. I think one of the other connectors is using this as well.

[the remark this answers is not in the document]
Comment (JingqiuDu) [reply stix_shifter_modules/azure_sentinel/stix_translation/json/to_stix_map.json:93]: aws_athena. and qradar both have `x-ibm-finding.description`

File: stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json [hunk 3]
```diff
@@ -152,10 +136,40 @@
...
+
+  "x-ibm-finding": {
+    "fields": {
+      "name": ["title"],
+      "description": ["description"],
+      "severity": ["severity"],
+      "start": ["createdDateTime"],
+      "end": ["closedDateTime"],
+      "finding_type": ["category"],
+      "src_ip_ref.value": ["networkConnections.natSourceAddress"],
```
[the remark this answers is not in the document]
Comment (JingqiuDu) [reply stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json:161]: fixed, see https://github.com/opencybersecurityalliance/stix-shifter/blob/59997e2197fbe07a524f8797d3b11677afc336d9/stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json#L10

File: stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json [hunk 4]
```diff
@@ -152,10 +136,40 @@
...
+      "dst_ip_ref.value": ["networkConnections.natDestinationAddress"],
+      "src_os_ref.name": ["hostStates.os"],
+      "dst_application_ref.name": ["cloudAppStates.destinationServiceName"],
+      "src_geolocation": ["networkConnections.sourceLocation"],
+      "dst_geolocation": ["networkConnections.destinationLocation"],
+      "src_application_user_ref.user_id":["userStates.aadUserId"],
```
[the remark this answers is not in the document]
Comment (JingqiuDu) [reply stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json:167]: user-account new field added
https://github.com/opencybersecurityalliance/stix-shifter/blob/59997e2197fbe07a524f8797d3b11677afc336d9/stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json#L67

File: stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json [hunk 5]
```diff
@@ -92,58 +92,42 @@
...
       "malwareStates.family": ["malwareStates.family"],
       "malwareStates.name": ["malwareStates.family"],
       "malwareStates.severity": ["malwareStates.family"],
       "malwareStates.wasRunning": ["malwareStates.family"],
-      "networkConnections.destinationLocation": ["networkConnections.destinationLocation"],
       "networkConnections.direction": ["networkConnections.direction"],
       "networkConnections.domainRegisteredDateTime": ["networkConnections.domainRegisteredDateTime"],
       "networkConnections.localDnsName": ["networkConnections.localDnsName"],
-      "networkConnections.natDestinationAddress": ["networkConnections.natDestinationAddress"],
       "networkConnections.natDestinationPort": ["networkConnections.natDestinationPort"],
```
[the remark this answers is not in the document]
Comment (JingqiuDu) [reply stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json:128]: added, https://github.com/opencybersecurityalliance/stix-shifter/blob/59997e2197fbe07a524f8797d3b11677afc336d9/stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json#L22

File: stix_shifter_modules/azure_sentinel/stix_translation/json/to_stix_map.json (hunk 1 above)
Comment (JingqiuDu) [reply stix_shifter_modules/azure_sentinel/stix_translation/json/to_stix_map.json]: fixed, see https://github.com/opencybersecurityalliance/stix-shifter/blob/59997e2197fbe07a524f8797d3b11677afc336d9/stix_shifter_modules/azure_sentinel/stix_translation/json/to_stix_map.json#L39 

https://github.com/opencybersecurityalliance/stix-shifter/blob/59997e2197fbe07a524f8797d3b11677afc336d9/stix_shifter_modules/azure_sentinel/stix_translation/json/to_stix_map.json#L2

**Comment by mdazam1942** [comment]
@JingqiuDu can you please resolve the conflicts?

File: stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json [hunk 6]
```diff
@@ -1,7 +1,15 @@
 {
   "ipv4-addr": {
     "fields": {
-      "value": ["networkConnections.sourceAddress", "networkConnections.destinationAddress", "hostStates.publicIpAddress", "hostStates.privateIpAddress", "userStates.logonIp"]
+      "value": [
+        "networkConnections.sourceAddress",
+        "networkConnections.destinationAddress",
+        "hostStates.publicIpAddress",
+        "hostStates.privateIpAddress",
+        "userStates.logonIp",
```
Comment (mdazam1942) [inline stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json:9]: remove "hostStates.publicIpAddress", "hostStates.privateIpAddress", "userStates.logonIp" from `"ipv4-addr"` and add those as `x-msazure-sentinel-alert` custom objects

File: stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json [hunk 7]
```diff
@@ -56,15 +64,15 @@
...
+      "user_id": ["userStates.accountName", "processes.accountName", "userStates.aadUserId"],
       "account_login": ["userStates.logonId"],
       "account_type": ["userStates.userAccountType"],
       "account_last_login": ["userStates.logonDateTime"]
     }
   },
   "software": {
     "fields": {
-      "name": ["vendorInformation.provider", "networkConnections.applicationName"],
+      "name": ["vendorInformation.provider", "networkConnections.applicationName", "hostStates.os"],
```
Comment (mdazam1942) [inline stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json:75]: remove `"networkConnections.applicationName", "hostStates.os"` from "name" and move them to `x-msazure-sentinel-alert`

File: stix_shifter_modules/azure_sentinel/stix_translation/json/to_stix_map.json [hunk 8]
```diff
@@ -30,19 +26,32 @@
...
+    "category": [
+      {
+      "key": "x-ibm-finding.finding_type",
+      "object": "finding"
+      },
+      {
+      "key": "x-oca-event.category",
+      "object": "event"
+      }
+    ],
```
Comment (mdazam1942) [inline stix_shifter_modules/azure_sentinel/stix_translation/json/to_stix_map.json:38]: Looks redundant to map `category` to both x-ibm-finding and x-oca-event object

Comment (delliott90) [reply stix_shifter_modules/azure_sentinel/stix_translation/json/to_stix_map.json:38]: Yeah, it should be removed from the finding object and left with the event object.

File: stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json (hunk 6 above)
Comment (JingqiuDu) [reply stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json]: removed

**Comment by JingqiuDu** [comment]
> @JingqiuDu can you please resolve the conflicts?

conficts resovled

File: stix_shifter_modules/azure_sentinel/stix_translation/json/to_stix_map.json (hunk 8 above)
Comment (JingqiuDu) [reply stix_shifter_modules/azure_sentinel/stix_translation/json/to_stix_map.json]: fixed

File: stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json (hunk 7 above)
Comment (JingqiuDu) [reply stix_shifter_modules/azure_sentinel/stix_translation/json/from_stix_map.json]: fixed

File: stix_shifter_modules/azure_sentinel/stix_translation/json/to_stix_map.json [hunk 9]
```diff
@@ -67,13 +71,17 @@
         "cybox": false
       },
       {
-        "key": "x-msazure-sentinel-alert.createddatetime",
-        "object": "alert"
+        "key": "x-ibm-finding.createddatetime",
+        "object": "finding"
+      },
+      {
+        "key": "x-oca-eventg.created",
```
Comment (pcoccoli) [inline stix_shifter_modules/azure_sentinel/stix_translation/json/to_stix_map.json:78]: Typo in `eventg`

**Review by mdazam1942 — APPROVED** [review]

**Review by delliott90 — APPROVED** [review]

**Pull request merged**